### DISCUSSION TITLE
Using GetSystemMetrics() in App.cpp

### DISCUSSION THREAD

**Discussion by fShen21** [microsoft/react-native-windows#9047] [Q&A] [answered]
Hello,

forgive me if this is common knowledge or has already been answered, I'm new to RNW/C++ and I couldn't find a solution anywhere.

I'm trying to set the size of the starting screen of the official RNW-demo application by using GetSystemMetrics(SM_CYSCREEN/CXSCREEN) in App.cpp, so I can get both width and height of the screen it is running on. I want the application to start in a screen-sized window on any machine it is running on.

The GetSystemMetrics - function is not recognized in App.cpp. I tried including  #pragma comment(lib, "user32.lib") together with <winuser.h>, <windows.h> and a couple of others. the function just won't work.
On the other hand, I tested it in a simple Visual Studio C++ console project, where I just need to include <winuser.h> to get it to run.

Is what I'm trying even possible?

Here is what it basically looks like:
![image](https://user-images.githubusercontent.com/93655384/140059722-84136d2c-5c65-4bf3-bdda-36d3c1bef734.png)

**Comment by chrisglein** [comment, marked as answer]
`GetSystemMetrics` is a desktop Win32 API, whereas RNW apps are UWP apps (see more [here](https://microsoft.github.io/react-native-windows/docs/native-code#win32-desktop-apps-vs-rnw-apps)). So you'd want the UWP equivalent API for that kind of device information. You can dig in maybe [here](https://docs.microsoft.com/en-us/uwp/api/windows.graphics.display.displayinformation.screenheightinrawpixels?view=winrt-22000) or [here](https://docs.microsoft.com/en-us/windows/win32/hidpi/high-dpi-desktop-application-development-on-windows).

However if we back up to your stated scenario:

> I want the application to start in a screen-sized window on any machine it is running on.

I think what you're looking for is simpler than all of that. Look up documentation on how to create a fullscreen UWP app, such as [this](https://docs.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.applicationview.tryenterfullscreenmode?view=winrt-22000)
